packages/amplify-e2e-core/src/categories/api.ts

```diff
@@ -609,19 +617,20 @@ export function updateRestApi(cwd: string, settings: Partial<typeof updateRestAp
     default:
       throw new Error(`updateOperation ${completeSettings.updateOperation} is not implemented`);
   }
-  chain.wait('Restrict API access').sendNo().wait('Do you want to add another path').sendNo().wait('Successfully updated resource');
+  chain.wait('Restrict API access').sendNo().wait('Do you want to add another path').sendNo()
+    .wait('Successfully updated resource');
   return chain.runAsync();
 }
 
 const allAuthTypes = ['API key', 'Amazon Cognito User Pool', 'IAM', 'OpenID Connect'];
 
-export function addApi(projectDir: string, settings?: any) {
+export function addApi(projectDir: string, settings?: any, requireAuthSetup = true) {
```

Comment:
could `requireAuthSetup` be part of `settings` ?

Reply:
Yes but because settings is untyped it's very difficult to figure out what's going on there and there's some weird stuff where we iterate over the keys of the object and I just didn't want to touch it too much

Reply:
1. I didn't notice this is e2e codebase.
2. It seems that `settings` is really `authTypesWithConfiguration` by looking at https://github.com/aws-amplify/amplify-cli/blob/aaf22103548a329b3092f52736c28c9b5c5b8b8b/packages/amplify-e2e-core/src/categories/api.ts#L629 .

Reply:
changed settings to `authTypesSettings` and added a slightly better type for it (`Record<string, $TSAny>`)